packages/desktop/src/services/FileDownloadServiceElectron.ts

```diff
@@ -66,7 +68,35 @@ export default class FileDownloadServiceElectron
         onProgress?: (transferredBytes: number) => void,
         destination?: string
     ): Promise<DownloadResult> {
-        let downloadUrl;
+        let downloadUrl: string;
+
+        if (fileInfo.path.endsWith(".zarr")) {
+            const isS3Path = this.isS3Url(fileInfo.path);
+            if (isS3Path) {
+                return this.downloadS3Directory(
+                    fileInfo,
+                    downloadRequestId,
+                    onProgress,
+                    destination
+                );
+            }
+
+            const isLocal = await this.isLocalPath(fileInfo.path);
+            if (isLocal) {
```

Comment:
Rather than checking for S3 specifically could we hope `zarrita` supports other cloud infra like Azure or whatever too or is that definitely not possible? If so, seems like this could be simpler and more versatile
Example:
```
const isLocalFile = await this.isLocalPath(fileInfo.path);
if (isLocalFile) {
    return this.copyLocalZarrDirectory(fileInfo, downloadRequestId, onProgress, destination);
}
return this.downloadS3Directory(fileInfo, downloadRequestId, onProgress, destination);
```

Reply:
Zarrita does not have any writing functions to save/write out the zarr files so we have to deal directly with where the file is stored. I think this means that we have to add support for each type of storage. The only test data I have right now is S3 but if we move towards other storage locations we can open a ticket for them and add individual support?  I don't really see a way to universalize directory download, unless there's some dependency we could use that I don't know about...